cfme/cloud/instance/__init__.py

```diff
@@ -176,318 +347,207 @@ def find_quadicon(self, *args, **kwargs):
+        else:
+            view = navigate_to(self, 'AllForProvider')
+            view.toolbar.view_selector.select('List View')
+            try:
+                row = view.paginator.find_row_on_pages(view.entities.table, name=self.name)
+            except NoSuchElementException:
+                raise InstanceNotFound('Failed to find instance in table: {}'.format(self.name))
+            row[0].check()
 
-    :param template_name: The image/template name to select
-    :param provider: Provider where the image/template resides
-    :return: none
-    """
-    logger.debug('Selecting an image {} from provider {} for provisioning'
-                 .format(template_name, provider.name))
-    navigate_to(Instance, 'Provision')
-    template = image_select_form.template_table.find_row_by_cells({
-        'Name': template_name,
-        'Provider': provider.name
-    })
-    if template:
-        sel.click(template)
-        # In order to mitigate the sometimes very long spinner timeout, raise the timeout
-        with sel.ajax_timeout(90):
-            sel.click(form_buttons.FormButton("Continue", force_click=True))
-
-    else:
-        raise TemplateNotFound('Unable to find template "{}" for provider "{}"'.format(
-            template_name, provider.key))
+        # cancel is the kwarg, when true we want item_select to dismiss the alert, flip the bool
+        view.toolbar.power.item_select(kwargs.get('option'),
+                                       handle_alert=not kwargs.get('cancel', False))
 
 
 @navigator.register(Instance, 'All')
-class InstanceAll(CFMENavigateStep):
+class All(CFMENavigateStep):
+    VIEW = InstanceAllView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
-    def am_i_here(self):
-        return match_page(summary='All Instances')
-
     def step(self):
         self.prerequisite_view.navigation.select('Compute', 'Clouds', 'Instances')
+        self.view.sidebar.instances.tree.click_path('All Instances')
 
-        # use accordion
+    def resetter(self, *args, **kwargs):
         # If a filter was applied, it will persist through navigation and needs to be cleared
-        if sel.is_displayed(search_box.clear_advanced_search):
+        if self.view.entities.adv_search_clear.is_displayed:
             logger.debug('Clearing advanced search filter')
-            sel.click(search_box.clear_advanced_search)
-        accordion.tree('Instances', 'All Instances')
+            self.view.entities.adv_search_clear.click()
+        self.view.toolbar.reload.click()
 
 
 @navigator.register(Instance, 'AllForProvider')
-class InstanceAllForProvider(CFMENavigateStep):
+class AllForProvider(CFMENavigateStep):
+    VIEW = InstanceProviderAllView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
-    def am_i_here(self):
-        return match_page(summary='Instances under Provider "{}"'.format(self.obj.provider.name))
-
     def step(self, *args, **kwargs):
```

Comment:
Include/pass them in case in the future args are passed through navigation. Not really needed at the moment though.